tools/mason/MasonPublish.chpl

```diff
@@ -78,6 +79,42 @@ proc masonPublish(ref args: list(string)) throws {
       args.append(potentialPath);
     }
 
+    if createReg {
+      var pathReg = '';
+      var idx_flag = 0;
+      var idx_pathReg = 0;
+      var idx = 0;
+      for arg in args {
+        if arg != "mason" && arg != "publish" {
```

Comment:
A loop with this structure could catch an error if a user gave multiple paths (and requires a little less manual index tracking):

```chpl
  var registryPath: string;
  for i in 2..<args.size {
    // Find positional arguments
    if !args[i].startsWith('-') {
      if registryPath == '' {
        registryPath = args[i];
      } else {
        // Multiple positional arguments is an error
        throw new owned MasonError("mason publish --create-registry expects only one path");
      }
    }
  }
  if registryPath == '' {
    // No positional arguments is an error
    throw new owned MasonError("mason publish --create-registry expects a path");
  }
```